x11-packages/alacritty/build.sh

```diff
@@ -3,98 +3,79 @@ TERMUX_PKG_DESCRIPTION="A fast, cross-platform, OpenGL terminal emulator"
-	_version="${_version/^/}"
-	local _url="https://github.com/rust-windowing/$_name/archive/refs/tags/v$_version.tar.gz"
-	local _path="$TERMUX_PKG_CACHEDIR/$_name-v$_version.tar.gz"
-	termux_download "$_url" "$_path" SKIP_CHECKSUM
-	tar xf "$_path" -C "$TERMUX_PKG_SRCDIR"
-	mv "$_name-$_version" "$_name-source"
-}
-
-termux_step_pre_configure() {
+termux_step_configure() {
 	termux_setup_cmake
 	termux_setup_rust
+	cargo clean
+	cargo vendor
 
-	: "${CARGO_HOME:=$HOME/.cargo}"
-	export CARGO_HOME
-
-	__cargo_fetch_dep_source_for_rust_windowing "winit"
-	__cargo_fetch_dep_source_for_rust_windowing "glutin"
-
-	patch="$TERMUX_PKG_BUILDER_DIR/patch-root-Cargo.diff"
-	patch -p1 -d "$TERMUX_PKG_SRCDIR" < "$patch"
-
-	cat "$TERMUX_PKG_BUILDER_DIR"/winit-*.diff | patch -p1 -d "$TERMUX_PKG_SRCDIR/winit-source"
-	cat "$TERMUX_PKG_BUILDER_DIR"/glutin-*.diff | patch -p1 -d "$TERMUX_PKG_SRCDIR/glutin-source"
-
-	cargo update
+	for dir in ./vendor/rustix*; do
+		[ -d "$dir" ] || continue
+		if [ -d "$dir/src/backend/libc/shm" ] && \
+			[ -d "$dir/src/backend/linux_raw/shm" ]; then
+			rm -rf "$dir/src/backend/libc/shm/"*
+			cp "$dir/src/backend/linux_raw/shm/"* \
+				"$dir/src/backend/libc/shm/"
+		fi
+	done
 
-	rm -rf "$CARGO_HOME"/registry/src/index.crates.io-*/expat-sys-*
-	rm -rf "$CARGO_HOME"/registry/src/index.crates.io-*/freetype-sys-*
-	rm -rf "$CARGO_HOME"/registry/src/index.crates.io-*/rustix-*
-	rm -rf "$CARGO_HOME"/registry/src/index.crates.io-*/servo-fontconfig-sys-*
-	rm -rf "$CARGO_HOME"/registry/src/index.crates.io-*/x11rb-protocol-*
-	cargo fetch --target "${CARGO_TARGET_NAME}"
+	if grep -q '^\[patch\.crates-io\]' Cargo.toml; then
+		sed -i '/^\[patch.crates-io\]/,$d' Cargo.toml
+	fi
 
-	local crate
-	for crate in {{expat,freetype,servo-fontconfig}-sys,rustix,x11rb-protocol}; do
-		local patch="$TERMUX_PKG_BUILDER_DIR/${crate}.diff"
-		local dir
-		for dir in "$CARGO_HOME"/registry/src/index.crates.io-*/"${crate}"-*; do
-			local _crate_name
-			_crate_name=$(basename "$dir")
-			# shellcheck disable=SC2295
-			if [[ ! "${_crate_name#$crate-}" =~ ^[0-9] ]]; then
-				continue
-			fi
-			if [[ "$crate" == 'rustix' ]]; then
-				rm -rf "$dir"/src/backend/libc/shm/*
-				cp "$dir"/src/backend/linux_raw/shm/* "$dir"/src/backend/libc/shm/
-			fi
-			echo "Applying patch for '$crate'"
-			patch -p1 -d "$dir" < "${patch}"
-		done
+	patch_lines=""
+	for patchfile in "$TERMUX_PKG_BUILDER_DIR"/*.vendor.diff; do
+		[ -e "$patchfile" ] || break
+		crate=$(basename "$patchfile" .vendor.diff)
+		dir="./vendor/$crate"
+		if [ ! -d "$dir" ]; then
+			echo "No vendor dir for $crate"
+			exit 1
+		fi
+		sed "s|@TERMUX_PREFIX@|${TERMUX_PREFIX}|g" "$patchfile" | \
+			patch --fuzz=0 -p1 -d "$dir"
+		patch_lines="${patch_lines}${crate} = { path = \"$dir\" }\n"
 	done
+
+	if [ -n "$patch_lines" ]; then
+		printf "\n[patch.crates-io]\n$patch_lines" >> Cargo.toml
+	fi
```

Comment:
- `$patch_lines`, `$patchfile`, `$crate` and `$dir` should be `local`
- `$patchfile` should probably be named `patch_file` for consistency.
- The `$crate` assignment should be quoted.
  - Sidenote, why are we using `basename` like that?
  I've never seen that form of it[^1] being used.
[^1]: `basename` "form 2" invocation - https://man.archlinux.org/man/basename.1#basename~2
- Use `[[` bash test `]]`, since our build scripts are written in Bash and it's less ambiguous.

```suggestion
	local patch_lines="" patch_file crate dir
	for patch_file in "$TERMUX_PKG_BUILDER_DIR"/*.vendor.diff; do
		[[ -e "$patch_file" ]] || break
		crate="$(basename "$patch_file" .vendor.diff)"
		dir="./vendor/$crate"
		if [[ ! -d "$dir" ]]; then
			echo "No vendor dir for $crate"
			exit 1
		fi
		sed "s|@TERMUX_PREFIX@|${TERMUX_PREFIX}|g" "$patch_file" | \
			patch --fuzz=0 -p1 -d "$dir"
		patch_lines="${patch_lines}${crate} = { path = \"$dir\" }\n"
	done

	if [[ -n "$patch_lines" ]]; then
		printf "\n[patch.crates-io]\n$patch_lines" >> Cargo.toml
	fi
```

We also don't *usually* specify `--fuzz` for `patch`, but I think it makes sense to here.